### PR TITLE
[Profile] internal log

### PR DESCRIPTION
**Pull request by jaeyun-jung**
add internal profile log function.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_profile.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+nns_profile_end (const gchar * name)
+{
+  nns_profile_task_s *task;
+  gint64 time_diff, average;
+
+  task = g_hash_table_lookup (g_nns_profile_task, name);
+  g_assert (task);
+
+  time_diff = g_get_monotonic_time () - task->start;
+  task->total += time_diff;
```
Comment (wooksong) [inline gst/nnstreamer/nnstreamer_profile.c:82]: Even though using averages, I am not sure that time-diff between two user-level code lines could represent the software-stack overhead. For Android, need to investigate https://developer.android.com/topic/performance/tracing/custom-events. For the other linux-based platforms, need to use perf API. 

Anyway as a fallback for those profiling framework time-diff would be one of our options.

File: gst/nnstreamer/nnstreamer_profile.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * @brief Function to set start time.
+ */
+void
+nns_profile_start (const gchar * name)
+{
+  nns_profile_task_s *task;
+
+  if (g_nns_profile_task == NULL) {
+    g_nns_profile_task =
```
Comment (dongju-chae) [inline gst/nnstreamer/nnstreamer_profile.c:54]: Q: Is it okay not to consider any locking mechanism to access this global hash table?

File: gst/nnstreamer/nnstreamer_profile.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  task = g_hash_table_lookup (g_nns_profile_task, name);
+  if (task == NULL) {
+    task = g_new0 (nns_profile_task_s, 1);
+    g_assert (task);
+    g_assert (g_hash_table_insert (g_nns_profile_task, g_strdup (name), task));
+  }
+
+  task->start = g_get_monotonic_time ();
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_profile.c:66]: This appears too heavy for a profile/instrumentation function.

**Comment by jaeyun-jung** [comment]
plz ignore this.

**Pull request closed without merging**